OpenRA.Mods.Common/Effects/FloatingSprite.cs

```diff
@@ -0,0 +1,98 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	class FloatingSprite : IEffect, ISync, ISpatiallyPartitionable
+	{
+		readonly WDist[] speed;
+		readonly WDist[] gravity;
+		readonly Animation anim;
+
+		readonly bool visibleThroughFog;
+		readonly int turnRate;
+		readonly int randomRate;
+		readonly string palette;
+
+		[Sync]
```

Comment:
Since this is just a decorative particle, should we care much about checking for `Sync` here?

Equally, should we use the cosmetic RNG rather than the shared RNG?

Reply:
Yes, of course. Well spotted.